fs/span-manager/span_manager.go

```diff
@@ -442,16 +380,75 @@ func (m *SpanManager) resolveSpanFromCache(s *span, offsetStart, size compressio
 	return nil, ErrSpanNotAvailable
 }
 
-func (m *SpanManager) fetchSpan(buf []byte, spanID compression.SpanID) error {
+// fetchAndCacheSpan fetches a span, uncompresses it and caches the uncompressed
+// span content.
+func (m *SpanManager) fetchAndCacheSpan(spanID compression.SpanID) ([]byte, error) {
+	// fetch compressed span
+	compressedBuf, err := m.fetchSpanWithRetries(spanID)
```

Comment:
I think one subtle difference between this and the previous version is that since we only need `compressedBuf` until `m.uncompressSpan`, the old version didn't escape the function and therefore could be allocated on the stack. Since this `[]byte` is being returned from another function, it causes a heap allocation. I'm not sure how much that really matters, but we should keep that in mind the next time we're taking a look at performance and memory usage.

Reply:
got it. So the difference is mainly on "return compressedBuf from `fetchSpanWithRetries` v.s. pass compressedBuf from caller into `fetchSpanWithRetries`". 

Also notice we are already returning (uncompressed, which should be larger) buffer (instead of pass in)  in `uncompressSpan`.

https://github.com/awslabs/soci-snapshotter/pull/446 will clean up several wrappers on span operations, which should make it easier to compare and unify how to take `[]byte buffer` (pass in or return).